floodviz/map_utils.py

```diff
@@ -41,37 +45,77 @@ def site_dict(site_list, url_prefix):
         else:
             break
 
-    # make a list of dicts from data
-    fields = data[0].split('\t')
+    keep_fields = [
+        'site_no',
+        'station_nm',
+        'dec_long_va',
+        'dec_lat_va',
+        'huc_cd'
+    ]
+
+    # make a list of dicts from data, remove any aberrant spaces
+    fields = [field.strip() for field in data[0].split('\t')]
     dnice = []
     for line in data[2:]:
-        line = line.split('\t')
+        line = [item.strip() for item in line.split('\t')]
         line_dict = dict(zip(fields, line))
-        dnice.append(line_dict)
+        try:
+            filtered_line_dict = {k: line_dict[k] for k in keep_fields}
+        except KeyError:
```

Comment:
If you are going to raise KeyError, need to reflect that in the function comments. However, I think I would just use line_dict.get(k, '') and default to nothing. Or we could just not return data for that line. It's fine to leave this as is for now, but something to consider.